### PR TITLE
feat(GuildIterator): fill `approximate_member_count` and `approximate_presence_count` fields when fetching Guilds

### PR DESCRIPTION
**Pull request by Victorsitou**
## Summary

Guilds fetched with `Client.fetch_guilds` now have `approximate_member_count` and `approximate_presence_count` fields filled.

https://github.com/discord/discord-api-docs/pull/5628

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by Skelmis — COMMENTED** [review]
I can't think of the use case for this param, but all other parameters to this HTTP method can be passed when this method is used in iterators. Shouldn't `with_counts` also be specified as an option within `GuildIterator`

**Comment by Victorsitou** [comment]
> I can't think of the use case for this param, but all other parameters to this HTTP method can be passed when this method is used in iterators. Shouldn't `with_counts` also be specified as an option within `GuildIterator`

If I remember correctly, we decided to keep it always as `True`.

**Comment by Skelmis** [comment]
> > I can't think of the use case for this param, but all other parameters to this HTTP method can be passed when this method is used in iterators. Shouldn't `with_counts` also be specified as an option within `GuildIterator`
> 
> If I remember correctly, we decided to keep it always as `True`.

I see, thanks.

**Review by Skelmis — APPROVED** [review]

**Comment by shiftinv** [comment]
@Skelmis brings up a valid point though. I don't think there are situations where you don't want this to be set to `True`, but it also wouldn't hurt to make it configurable for `fetch_guild` and `fetch_guilds` just in case.

**Comment by shiftinv** [comment]
the API docs PR for this got merged, could you update this one as discussed above?

**Comment by Victorsitou** [comment]
Done. 6046af92957d875467c03775f4f9a2d4642df333

File: changelog/892.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Added ``with_counts`` field to :meth:`Client.fetch_guild` and :meth:`Client.fetch_guilds`, used to determine whether or not to include the guild's member count and presence information.
```
Comment (shiftinv) [inline changelog/892.feature.rst:1]:
```suggestion
Added ``with_counts`` parameter to :meth:`Client.fetch_guild` and :meth:`Client.fetch_guilds`, used to determine whether to include the guild's member count and presence information.
```

**Review by shiftinv — APPROVED** [review]
lgtm, thanks!

File: disnake/client.py [hunk 2]
```diff
@@ -1772,6 +1773,9 @@ def fetch_guilds(
             Retrieve guilds after this date or object.
             If a datetime is provided, it is recommended to use a UTC aware datetime.
             If the datetime is naive, it is assumed to be local time.
+        with_counts: :class:`bool`
+            Whether to include approximate member and presence counts for the guilds.
+            Defaults to ``True``.
```
Comment (shiftinv) [inline disnake/client.py:1778]:
```suggestion
            Defaults to ``True``.

            .. versionadded:: 2.10
```

File: disnake/client.py [hunk 3]
```diff
@@ -1829,6 +1833,9 @@ async def fetch_guild(self, guild_id: int, /) -> Guild:
         ----------
         guild_id: :class:`int`
             The ID of the guild to retrieve.
+        with_counts: :class:`bool`
+            Whether to include approximate member and presence counts for the guild.
+            Defaults to ``True``.
```
Comment (shiftinv) [inline disnake/client.py:1838]:
```suggestion
            Defaults to ``True``.

            .. versionadded:: 2.10
```

**Review by shiftinv — COMMENTED** [review]
oops, noticed one more thing

**Comment by Victorsitou** [comment]
oops

**Pull request merged**